tests/Monolog/RegistryTest.php

```diff
@@ -59,4 +59,96 @@ public function hasLoggerProvider()
             ),
         );
     }
+
+    /**
+     * @covers Monolog\Registry::clear
+     */
+    public function testClearClears()
+    {
+        Registry::addLogger(new Logger('test1'), 'log');
+        Registry::clear();
+
+        $this->setExpectedException('\InvalidArgumentException');
+        Registry::getInstance('log');
+    }
+
+    /**
+     * @dataProvider removedLoggerProvider
+     * @covers Monolog\Registry::addLogger
+     * @covers Monolog\Registry::removeLogger
+     */
+    public function testRemovesLogger($loggerToAdd, $remove)
+    {
+        Registry::addLogger($loggerToAdd);
+        Registry::removeLogger($remove);
+
+        $this->setExpectedException('\InvalidArgumentException');
```

Comment:
I set the exception in here instead of as an annotation because it's a test implementation detail, not part of the interface we're testing.